src/mono/mono/mini/method-to-ir.c

```diff
@@ -7630,7 +7700,7 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				}
 			}
 
-			check_method_sharing (cfg, cmethod, &pass_vtable, &pass_mrgctx);
+			pass_mrgctx = need_mrgctx_arg (cfg, cmethod);
```

Comment:
@vargaz Is this going to pass an mrgctx to icalls or pinvokes now after the change to `mini_method_needs_mrgctx` ?

Reply:
No, there are additional checks:
```
	if (mini_method_needs_mrgctx (cmethod)) {
		if (mono_method_is_generic_sharable_full (cmethod, TRUE, TRUE, TRUE)) {
			pass_mrgctx = TRUE;
		} else {
			if (cfg->gsharedvt && mini_is_gsharedvt_signature (mono_method_signature_internal (cmethod)))
				pass_mrgctx = TRUE;
		}
	}
```

This code could definitely use some refactoring through.